lang/syn/src/idl/defined.rs

```diff
@@ -496,7 +496,7 @@ pub fn gen_idl_type(
                 use crate::parser::context::CrateContext;
                 use quote::ToTokens;
 
-                let source_path = proc_macro2::Span::call_site().source_file().path();
+                let source_path = proc_macro2::Span::call_site().file();
```

Comment:
The documentation in https://github.com/dtolnay/proc-macro2/blob/24bbf16d9df01d5f7d9ac39bdfbaea85f4c194fb/src/lib.rs#L467-L478 states for `.file`:

> This might not correspond to a valid file system path. It might be remapped, or might be an artificial path such as `"<macro expansion>"`.

Shouldn't `.local_file()` be used instead?

Reply:
my mistake, good catch! fixed!